src/main/java/Duke/CommandList.java

```diff
@@ -0,0 +1,133 @@
+package Duke;
+
+import Task.Task;
+import Task.ToDo;
+import Task.Deadline;
+import Task.Event;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class CommandList {
+
+    /**
+     * Execute different actions based on user's input taken from Parser.
+     *
+     * @param input command to execute
+     * @param parser Parser object to terminate program if necessary
+     * @param list Arraylist of Tasks to manipulate
+     * @param storage Storage object to load/store list of Tasks
+     */
+    public void execute(String input, Parser parser, ArrayList<Task> list, Storage storage) {
```

Comment:
I find this function to be long and cumbersome to read. Perhaps you could include some line breaks, or extract some parts of the code into a separate function?